testing/e2e/api_version/client_api_2023_06_01.go

```diff
@@ -0,0 +1,250 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build e2e
+
+package api_version
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"io"
+	"math"
+	"net/http"
+
+	"github.com/elastic/fleet-server/v7/pkg/api/versions/2022_06_01/api"
+
+	"github.com/elastic/fleet-server/v7/version"
+	"github.com/stretchr/testify/suite"
+)
+
+// ClientAPITester20230601 provides methods to test API endpoints
+type ClientAPITester20230601 struct {
+	suite.Suite
+
+	ctx      context.Context
+	client   *http.Client
+	endpoint string
+}
+
+func NewClientAPITester20230601(suite suite.Suite, ctx context.Context, client *http.Client, endpoint string) *ClientAPITester20230601 {
+	return &ClientAPITester20230601{
+		suite,
+		ctx,
+		client,
+		endpoint,
+	}
+}
+
+func (tester *ClientAPITester20230601) getAPIClient(withRequestEditorFn api.RequestEditorFn) (api.ClientWithResponsesInterface, error) {
+	return api.NewClientWithResponses(tester.endpoint, api.WithHTTPClient(tester.client), api.WithRequestEditorFn(func(ctx context.Context, req *http.Request) error {
+		req.Header.Set("elastic-api-version", "2023-06-01")
+
+		if withRequestEditorFn != nil {
+			return withRequestEditorFn(ctx, req)
+		}
+
+		return nil
+	}))
+}
+
+// TestStatus tests the status endpoint, if an apiKey is given the authenticated response if verfied.
+func (tester *ClientAPITester20230601) TestStatus(apiKey string) {
+	client, err := tester.getAPIClient(nil)
+	tester.Require().NoError(err)
+
+	var resp *api.StatusResponse
+	if apiKey != "" {
+		resp, err = client.StatusWithResponse(tester.ctx, &api.StatusParams{}, func(ctx context.Context, req *http.Request) error {
+			req.Header.Set("Authorization", "ApiKey "+apiKey)
+			return nil
+		})
+	} else {
+		resp, err = client.StatusWithResponse(tester.ctx, &api.StatusParams{})
+	}
+	tester.Require().NoError(err)
+	tester.Require().Equal(http.StatusOK, resp.StatusCode())
+	tester.Require().NotEmpty(resp.JSON200.Name)
+	tester.Require().NotEmpty(resp.JSON200.Status)
+	if apiKey != "" {
+		tester.Require().NotNil(resp.JSON200.Version)
+	}
```

Comment:
Nit. Use Assert for non-critical assertions so they are all executed.
```suggestion
	tester.Assert().Equal(http.StatusOK, resp.StatusCode())
	tester.Assert().NotEmpty(resp.JSON200.Name)
	tester.Assert().NotEmpty(resp.JSON200.Status)
	if apiKey != "" {
		tester.Assert().NotNil(resp.JSON200.Version)
	}
```